plugin/core/sessions.py

```diff
@@ -1969,6 +1969,12 @@ def send_request_task(self, request: Request) -> Promise:
         self.send_request_async(request, resolver, lambda x: resolver(Error.from_lsp(x)))
         return promise
 
+    def send_request_task_2(self, request: Request) -> Tuple[Promise, int]:
```

Comment:
Currently the request_id is stored on the session,
but if would store the id on the `Request` class the `send_request_task_2` could be removed. 
I did a quick experiment here -> https://github.com/sublimelsp/LSP/commit/d2661904a03c894afaab82b96eb4fc61bccfcf4f

Reply:
I did think about it like that but implementing it without hacks seemed like too much work. I don't like the approach with a global. Globals are rarely a correct way to do things. Also it's not thread-safe since we are creating `Request` instances both on the main and async thread.

Reply:
You know what, this function is not that bad.
It is the same as the above function, not that complicated, and the only difference is the return type.

If you find a way to remove it, good, if not, it still ok.